### PR TITLE
Adopt bunchee as bundler

### PR DESCRIPTION
**Pull request by huozhi**
Use [bunchee](https://github.com/huozhi/bunchee) as bundler for cjs/esm output

* replace ncc
* minify all assets
* better watch mode for all assets

reasons that didn't use common bundlers:

* microbundle - doesn't support esm interop on cjs output
* esbuild - doesn't support es5 output, might have browser compatible issues

in long term, would still love to adopt the trending bundler from community

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,22 +3,19 @@
...
   "repository": "vercel/swr",
   "homepage": "https://swr.vercel.app",
   "license": "MIT",
   "scripts": {
-    "build": "npm run build:esm && npm run build:cjs",
-    "build:cjs": "ncc build src/index.ts -o dist -m -e react",
-    "build:esm": "tsc --module ES6 --outDir esm",
-    "watch": "tsc --watch",
-    "watch:esm": "tsc --module ES6 --outDir esm --target esnext --watch",
+    "build": "bunchee src/index.ts --no-sourcemap -m",
```
Comment (huozhi) [inline package.json:16]: Unsure thing about whether to keep sourcemap here. Might increase the debug difficulty on production with minified code only for users.

File: src/use-swr.ts [hunk 2]
```diff
@@ -584,10 +584,7 @@ function useSWR<Data = any, Error = any>(
     const softRevalidate = () => revalidate({ dedupe: true })
 
     // trigger a revalidation
-    if (
-      isUpdating ||
-      willRevalidateOnMount()
-    ) {
+    if (isUpdating || willRevalidateOnMount()) {
```
Comment (huozhi) [inline src/use-swr.ts:587]: missing lines for prettier

File: package.json (hunk 1 above)
Comment (shuding) [reply package.json:16]: I think for now let's include it as it helps identifying the original code, but I doubt if anyone is using it.

**Comment by shuding** [comment]
CJS build is now 32% smaller, great job! 🎉

**Comment by shuding** [comment]
Published as `swr@1.0.0-beta.2` 🎉

**Pull request merged**